recipes/geographiclib/all/conanfile.py

```diff
@@ -132,12 +128,19 @@ def package(self):
         ]:
             rmdir(self, os.path.join(os.path.join(self.package_folder, folder)))
         rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        if not self.options.tools:
+            rmdir(self, os.path.join(self.package_folder, "sbin"))
+            bin_files = [it for it in os.listdir(os.path.join(self.package_folder, "bin")) if not it.endswith(".dll")]
+            for it in bin_files:
+                rm(self, it, os.path.join(self.package_folder, "bin"))
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "geographiclib")
         self.cpp_info.set_property("cmake_target_name", "GeographicLib::GeographicLib")
         self.cpp_info.set_property("pkg_config_name", "geographiclib")
-        self.cpp_info.libs = collect_libs(self)
+        suffix = "_d" if self.settings.build_type == "Debug" and is_msvc(self) else ""
+        suffix += "-i" if is_msvc(self) else ""
```

Comment:
Not sure about this suffix, as this is still causing the same issue on Windows.

Reply:
I'll enter in Windows to check it. Thank you for running it.

Reply:
According to upstream CMakeLists:

- Generic name is `GeographicLib` irrespective of major version.

- Postfix is `-i` if compiler is cl like (msvc & clang-cl) and lib is shared.

- Debug postfix is unpredictable if compiler is not cl-like, because there is a condition based on generator: https://github.com/geographiclib/geographiclib/blob/v2.3/CMakeLists.txt#L187-L193

I would keep `collect_libs()` in this recipe, or isolate this change in another PR. But you can expect regressions if you remove collect_libs, due to unpredictable naming and even difficulties to predict if CMake variable `MSVC` variable is true or not based on conan profile.

Reply:
I'm confused. @uilianries did you read what I wrote here? Lib name is unpredictable for conan in some cases (generator dependent), and clang-cl is likely broken now.